chimney/src/test/scala/io/scalaland/chimney/DslSpec.scala

```diff
@@ -531,6 +531,39 @@ class DslSpec extends WordSpec with MustMatchers {
       transformer1.transform mustBe Bar(1.0, 3, "abc")
       transformer2.transform mustBe Bar(6.0, 3, "abc")
     }
+
+    "transform from non-case class to case class" when {
+      import NonCaseDomain._
+
+      "support non-case classes inputs" in {
+        val source = new ClassSource("test-id", "test-name")
+        val target = source.transformInto[CasesTarget]
+
+        target.id mustBe source.id
+        target.name mustBe source.name
+      }
+
+      "support trait inputs" in {
+        val source: TraitSource = new TraitSourceImpl("test-id", "test-name")
+        val target = source.transformInto[CasesTarget]
+
+        target.id mustBe source.id
+        target.name mustBe source.name
+      }
+
+      "support java beans" in {
+        val source = new JavaBeanSource("test-id", "test-name")
+        val target = source
+          .into[CasesTarget]
+          .withFieldRenamed(_.getId, _.id)
```

Comment:
If we really want to support reading from Java bean sources, I think we should interpret `.getXXX` methods automatically. But I think it's separate issue: https://github.com/scalalandio/chimney/issues/71